### PR TITLE
openafs: Add support for Linux 5.0 to openafs_1_8

### PR DESCRIPTION
**Pull request by artemist**
###### Motivation for this change

OpenAFS 1.8.2 does has not yet merged support for Linux 4.20 or 5.0

###### Things done
This patch cherry-picks commits from the git repository, and includes
them in the patches/ directory. This also replaces the amalgamated Linux 4.20
patches I added earlier with the original outputs from `git show`.

This has been tested to build on Linux 4.4, 4.14, 4.20, and 5.0.

This does not change the derivations of openafs_1_8, only
linuxPackages_*.openafs_1_8.

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Assured whether relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by veprbl** [comment]
cc @spacefrogg

**Comment by artemist** [comment]
@GrahamcOfBorg build linuxPackages_5_0.openafs_1_8

**Comment by veprbl** [comment]
@artemist You might need to add yourself to known-users by opening a PR against https://github.com/NixOS/ofborg

@GrahamcOfBorg build linuxPackages_5_0.openafs_1_8

File: pkgs/servers/openafs/1.8/patches/current_kernel_time.patch [hunk 1]
```diff
@@ -0,0 +1,63 @@
+commit 3c454b39d04f4886536267c211171dae30dc0344
```
Comment (veprbl) [inline pkgs/servers/openafs/1.8/patches/current_kernel_time.patch]: This looks exactly like
http://git.openafs.org/?p=openafs.git;a=patch;h=aa80f892ec39e2984818090a6bb2047430836ee2
So we should be able to use fetchpatch after all.

File: pkgs/servers/openafs/1.8/patches/do_settimeofday.patch [hunk 2]
```diff
@@ -0,0 +1,60 @@
+commit aa80f892ec39e2984818090a6bb2047430836ee2
```
Comment (veprbl) [inline pkgs/servers/openafs/1.8/patches/do_settimeofday.patch]: This looks exactly like
http://git.openafs.org/?p=openafs.git;a=patch;h=aa80f892ec39e2984818090a6bb2047430836ee2
So we should be able to use fetchpatch after all.

Comment (artemist) [reply pkgs/servers/openafs/1.8/patches/do_settimeofday.patch]: It seems my problem with fetchpatch was assuming `nix-prefetch-url` would give me the right hashes. I've switched to fetchpatch.

Comment (veprbl) [reply pkgs/servers/openafs/1.8/patches/do_settimeofday.patch]: Yes, fetchpatch will apply some extra filtering to the patches, so the hash from the fixed-output derivation will often not match the original patches hash.

**Comment by veprbl** [comment]
@GrahamcOfBorg build linuxPackages_5_0.openafs_1_8

**Comment by artemist** [comment]
I can remove the patches from the new directory I created like I did in #56862, although IMHO it looks cleaner with all 3 patches in a directory (it was cleaner without the dir with 1.6 though)

**Comment by spacefrogg** [comment]
Looks good to me. I have no particular opinion on going with or without a patches/ directory.

**Review by veprbl — APPROVED** [review]
Works. Most kernels build:
```
2 package are marked as broken and were skipped:
linuxPackages_hardkernel_4_14.openafs_1_8 linuxPackages_hardkernel_latest.openafs_1_8

4 package failed to build:
linuxPackages_hardened.openafs_1_8 linuxPackages_latest-libre.openafs_1_8
linuxPackages_latest_hardened.openafs_1_8 linuxPackages_testing_hardened.openafs_1_8

10 package were build:
linuxPackages-libre.openafs_1_8 linuxPackages.openafs_1_8 linuxPackages_4_19.openafs_1_8
linuxPackages_4_20.openafs_1_8 linuxPackages_4_4.openafs_1_8 linuxPackages_4_9.openafs_1_8
linuxPackages_5_0.openafs_1_8 linuxPackages_latest_xen_dom0.openafs_1_8
linuxPackages_testing_bcachefs.openafs_1_8 linuxPackages_xen_dom0.openafs_1_8
```

**Comment by veprbl** [comment]
Thank you!

**Pull request merged**